pkg/providers/aws/provider_redis_metrics.go

```diff
@@ -153,7 +155,42 @@ func (r *RedisMetricsProvider) scrapeRedisCloudWatchMetricData(ctx context.Conte
 
 }
 
-func buildRedisMetricDataQuery(resourceID string) []*cloudwatch.MetricDataQuery {
-	//todo build this out
-	return nil
+func buildRedisMetricDataQuery(resourceID string, replicationGroups []*elasticache.ReplicationGroup,  metricTypes []providers.CloudProviderMetricType ) []*cloudwatch.MetricDataQuery {
+	var metricDataQueries []*cloudwatch.MetricDataQuery
+	for _, metricType := range metricTypes {
+		// Metrics are returned per node for ElastiCache
+
+		var foundCache *elasticache.ReplicationGroup
+		for _, c := range replicationGroups {
+			if *c.ReplicationGroupId == resourceID {
+				foundCache = c
+				break
+			}
+		}
+		// poll MemberCluster array for CacheClusterId
+		for _, cacheClusterId := range foundCache.MemberClusters{
+			//cast string pointer to string
+			cacheClusterIdString := "" + *cacheClusterId
+			metricDataQueries = append(metricDataQueries, &cloudwatch.MetricDataQuery{
+				// id needs to be unique, and is built from the metric name and type, and the CacheClusterId number
+				// the metric name is converted from camel case to snake case to allow it to be easily reused when exposing the metrics
+				Id: aws.String(metricType.PromethuesMetricName +"_"+ cacheClusterIdString[len(cacheClusterIdString)-3:]),
```

Comment:
We want to only return a single metric ID based on `PromethuesMetricName`, I am thinking we can either look to expose the `cacheClusterID` in the label `resourceID` or we only look to monitor the primary node. I think this would be worth a discussion, as it will somewhat change/diverge from the behavior of the Postgres Provider

Reply:
yea lets discuss it